src/storage/v2/replication/replication.hpp

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 Memgraph Ltd.
```

Comment:
what do you think about calling this file replication_state instead of replication since everything is about replication state

Reply:
Will do, I've put that on my list for a future PR.